DBSCE/Overview.md

```diff
@@ -0,0 +1,219 @@
+
+### Device Registration Client
+
+This is a pre-requisite for DBSC(E) to work.
+
+Device Registration Client is a process where the user registers the device with the IdP and is expected to be a once-in-a-lifetime operation.
+
+The device registration establishes trust between the device and a service that maintains a directory of all devices. This document does not cover the protocol of device registration, but it assumes that during device registration, some asymmetric keys are shared between the client and the service, typically a device key and some other keys necessary for the secure device communication. A client software component that performs the device registration is called a _device registration client_. As mentioned above, the key assumption in DBSC(E) is that device registration happened in a clean room environment, and it is the responsibility of the device owner to ensure this.
+
+One device registration client can manage multiple devices on the same physical device. There also can be multiple device registration clients on the same device. The device registration client can be owned and supported by:
+
+- Operating system - the device gets registered when the OS is installed.
+- Browser - the device gets registered when the browser is installed.
+- Device management software (MDM provider) - the device gets registered when the MDM is enrolled.
+- Third-party software vendor - the device gets registered according to the vendor rules.
+
+DBSC(E) aims to support most of these scenarios. It does not define the device registration protocol amd is only concerned with the keys generated in a "clean room" state and the management of the generated keys to prove device binding.
+
+### Local Key Helper
+
+DBSC(E) introduces the concept of `Local Key Helper` which can be mapped to the `TPM` or any `Key generation helper` for the consumer case.
+**Local Key Helper** is an integral part of the the **Device Registration Client**, a software interface responsible for the DBSC Key management. It can be Public or Private and is expected to be either shipped as a part of a given enterprise framework (with the IdP/OS) or can be installed by a provider in compliance with the protocol expanded below.
+
+From the deployment point of view there are two types of local key helpers: _well-known_(_private_) and _third party_(_public_)
+
+- _Public local key helper_ or _third party_: Can be accessed by any Identity Provider (IdP). Typically owned by a provider different from the IdP, communicates with the IdP through as defined in DBSC(E) protocol. A third party local key helper has a special deployment mechanism. We expect to loosely define the API and deployment mechanism in the upcoming specification, and will leave the specific implementation details to the vendors. However, we have provided examples of how this can look for a few platforms [below](#platform-examples).
```

Comment:
The intent was to leave space for implementations by other providers. Probably `loosely` isn't a correct word for the API especially. I can rephrase it to convey the same intent. For example, we provided platform specifics on how it can look but are we prescribing the specifics of a every local key helper? 

Let me redo this section to convey that:
- We added details of Local key helpers for windows/mac/android platforms (these are still *one* type per platform)
- Define what level of flexibility are we offering for any platform/Idp that wants to create their own local key helper/or use an external key generation service as their local key helper.

I agree that we should have specifics about:
* How and where browser should find `local key helpers`
* Which local key helper a browser will choose (the first)
* The API between the browser and local key helper

Reply:
I think there're 3 separate things here and the section combines all of them, thus causing confusion:

1. How does browser discover and manage local key helpers. This should be strictly defined. 
2. How does each specific public local key helper manage its keys and performs attestation. This one I agree is up to each local key helper provider to design, but it needs to be strictly defined by themselves for their customers, and the protocol should follow DBSC(E) semantics.
3. Categorization of local key helpers. Current categorization combines 2 aspects - whether key helper can be used by any IDP or just one (public vs private), and whether key helper comes pre-installed. Combining both in one category is confusing, I'd suggest splitting - focus on categorizing key helpers as public and private only, and later explain that some key helpers can come pre-installed by default. 

Based on above, added some rewording suggestions below in bold:

> **Local Key Helper** is an integral part of the the **Device Registration Client**, a software interface responsible for the DBSC Key management. *Local key helper* can be public or private and is expected to be either shipped as a part of a given enterprise framework (with the IdP/OS) or can be installed by a provider in compliance with the protocol expanded below. **DBSC(E) defines the details how browsers interact with local key helpers for each platform. Those details are outlined in KeyGeneration.md.** 

**There're two types of local key helpers: private and public**

> - _Public local key helper: **Those have a well-documented implementation and they can be used by any IDP.** Typically owned by a provider different from the IdP, communicates with the IdP as defined in DBSC(E) protocol. 

- _Private local key helper: **Can be only used by a specific IDP that owns the implementation.** It will have a private protocol to communicate with the IdP. 

**Additionally, some local key helpers are well-known and will be enabled by default. A browser knows how to activate a well-known key helper without needing of an additional admin-side configuration.**

Reply:
Made changes. Let us closely check, especially wording around protocol for public key helpers, I think we should add more details.